source/opt/fix_func_call_arguments.h

```diff
@@ -24,18 +24,20 @@ class FixFuncCallArgumentsPass : public Pass {
   FixFuncCallArgumentsPass() {}
   const char* name() const override { return "fix-for-funcall-param"; }
   Status Process() override;
-  // Query module whether function number is one
-  bool HasNoFunctionToCall();
-  // Replace function call argument of accesschain with memory variables
-  void ReplaceAccessChainFuncCallArguments(Instruction* func_call_inst,
-                                           Operand* operand,
-                                           Instruction* operand_inst,
-                                           Instruction* next_inst,
-                                           Instruction* var_insertPt,
-                                           unsigned operand_index);
+  // Whether module has only one function
+  bool ModuleHasASingleFunction();
+  // Copies from the memory pointed to by |operand_inst| to a new function scope
+  // variable created at [var_insertPt] before |func_call_inst|, and copies the
+  // value of the new variable back to the memory pointed to by |operand| after
+  // |funct_call_inst| at |next_inst|. Returns the id of the new variable.
```

Comment:
Part of the reason I wanted to remove `var_instetrPt` and `next_inst` is because you now need to call out the requirements for those variables, which are a little involved.  You'll need to mention that `var_insertPt` must be in the same function as `funct_call_inst` and must be in a position where it is legal to insert an variable.

You'll also have to mention that `next_inst` must post-dominate `funct_call_inst` and dominate all uses of the operand.

If you insist on keeping these two parameters, add asserts to make sure these conditions are met.  When these parameters have to have a particular relationship in order for the function to work correctly, it becomes easy for someone to get it wrong.  We want to identify these errors as early as possible.  Even though things are fine now, this is done for other people who might need to change this pass.

I would still prefer you compute these value in the function, making it easier for others to know how the function should be used.